### ISSUE TITLE
BlockStorage v2: Support Cinder "noauth" auth_strategy

### ISSUE THREAD

**Issue by tchughesiv** [gophercloud/gophercloud#502]
Here's my current approach:
https://github.com/gophercloud/gophercloud/compare/master...tchughesiv:cinder-noauth

Example of Creating a Service Client
```golang
	ao, err := openstack.AuthOptionsFromEnv()
	provider, err := openstack.UnAuthenticatedClient(ao)
	client, err := openstack.NewBlockStorageV2(provider, gophercloud.EndpointOpts{
		Region:         os.Getenv("OS_REGION_NAME"),
		CinderEndpoint: os.Getenv("CINDER_ENDPOINT"),
	})
```

This is already supported in the python client:
https://github.com/openstack/python-cinderclient/commit/60f92db7049b4f66e5198b86bfecc1029b6cdccd

**Comment by jtopjian** [comment]
@tchughesiv Would it be possible to implement this similar to swauth?

https://github.com/gophercloud/gophercloud/tree/master/openstack/objectstorage/v1/swauth

**Comment by tchughesiv** [comment]
@jtopjian don't see why not... let me give it a go in a separate branch.

**Comment by tchughesiv** [comment]
@jtopjian here's a first go add new route ... still no tests yet, but wanted initial feedback:
https://github.com/gophercloud/gophercloud/compare/master...tchughesiv:cinder-new-noauth

used like:
```golang
	ao, err := openstack.AuthOptionsFromEnv()
	client, err := noauth.NewBlockStorageV2(ao, noauth.EndpointOpts{
		CinderEndpoint: os.Getenv("CINDER_ENDPOINT"),
	})
```

**Comment by jtopjian** [comment]
@tchughesiv I think it's a good start and it's worth opening a PR. 

Two comments:

1. Can you provide a link to the cinder-api / server-side implementation of this?
2. I think `ProviderClient` should be taken out of the `NewBlockStorageV2` function. I understand why you would want to bundle them together, but for consistency, it should be broken out. You can always write an app-specific `CinderNoAuth` function that bundles the two.

**Comment by tchughesiv** [comment]
@jtopjian cinder api code:
https://github.com/openstack/cinder/blob/92980e4972356f16389539b4cc926d7b4ec1dc3c/cinder/api/middleware/auth.py#L117

PR coming soon... tks for feedback

**Comment by jrperritt** [comment]
closed by #503
